src/App.tsx

```diff
@@ -1,24 +1,161 @@
+        setErrorMessage(ErrorMessage.LOAD);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const handleAddTodo = (title: string) => {
+    if (title.length === 0) {
+      setErrorMessage(ErrorMessage.EMPTY);
+      setTimeout(() => setErrorMessage(ErrorMessage.NONE), 3000);
+    } else {
+      const newTodo = {
+        id: 0,
+        userId: USER_ID,
+        completed: false,
+        title,
+      };
+
+      setTempTodo(newTodo);
+      setErrorMessage(ErrorMessage.NONE);
+
+      addTodo(USER_ID, newTodo)
+        .then((todo) => {
+          setTodos((prevTodos) => {
+            return [...prevTodos, todo];
+          });
+        })
+        .catch(() => {
+          setErrorMessage(ErrorMessage.ADD);
+          setTimeout(() => setErrorMessage(ErrorMessage.NONE), 3000);
+        })
+        .finally(() => {
+          setTempTodo(null);
+        });
+    }
+  };
+
+  const handleDeleteTodo = useCallback(
+    (id: number) => {
+      setLoadingTodoId(id);
+
+      return deleteTodo(id)
+        .then(() => {
+          setTodos((prevTodos) => {
+            return prevTodos.filter((todo) => todo.id !== id);
+          });
+        })
+        .catch(() => {
+          setErrorMessage(ErrorMessage.DELETE);
+          setTimeout(() => setErrorMessage(ErrorMessage.NONE), 3000);
+        })
+        .finally(() => {
+          setTempTodo(null);
+          setLoadingTodoId(0);
+        });
+    },
+    [deleteTodo],
+  );
+
+  const handleClearCompleted = () => {
+    todos.filter((todo) => todo.completed).forEach((todo) => {
+      handleDeleteTodo(todo.id)
+        .then(() => setTodos(todos.filter(({ completed }) => !completed)));
+    });
+  };
+
+  const handleToggleTodo = (id: number) => {
+    const todoCheck = todos.find((todo) => todo.id === id) as Todo;
+
+    todoCheck.completed = !todoCheck?.completed;
+
+    const notChanged = todos.filter(todo => todo.id !== id);
+
+    setTodos(() => [...notChanged, todoCheck]);
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          activeTodosLength={visibleTodos.length}
+          onSubmit={handleAddTodo}
+        />
+
+        <TodoList
+          todos={visibleTodos}
+          tempTodo={tempTodo}
+          onDelete={handleDeleteTodo}
+          loadingTodoId={loadingTodoId}
+          onToggle={handleToggleTodo}
+        />
+
+        {todos.length && (
+          <Footer
+            sortType={sortType}
+            setSortType={setSortType}
+            activeTodosLength={visibleTodos.length}
```

Comment:
as said in the prop name it should show `active` todos count, i.e todos with `todo.completed === false`

Reply:
fixed